src/plugins/template/backend/ops/bitwise_not.cpp

```diff
@@ -0,0 +1,47 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "openvino/op/bitwise_not.hpp"
+
+#include "evaluate_node.hpp"
+#include "openvino/reference/bitwise_not.hpp"
+
+using namespace ov;
+
+template <element::Type_t ET>
+bool evaluate(const std::shared_ptr<ov::op::v13::BitwiseNot>& node,
+              ov::TensorVector& outputs,
+              const ov::TensorVector& inputs) {
+    using T = typename ov::element_type_traits<ET>::value_type;
+    ov::reference::bitwise_not(inputs[0].data<T>(), outputs[0].data<T>(), shape_size(inputs[0].get_shape()));
+    return true;
```

Comment:
To follow common approach for evaluates, consider to add tensor size checks and set_shape for the output tensor:
```suggestion
bool evaluate(const std::shared_ptr<ov::op::v13::BitwiseNot>& node,
              ov::TensorVector& outputs,
              const ov::TensorVector& inputs) {
    OPENVINO_ASSERT(inputs.size() == 1);
    OPENVINO_ASSERT(outputs.size() == 1);
    outputs[0].set_shape(inputs[0].get_shape());

    using T = typename ov::element_type_traits<ET>::value_type;
    ov::reference::bitwise_not(inputs[0].data<T>(), outputs[0].data<T>(), shape_size(inputs[0].get_shape()));
    return true;
```

Reply:
I think is not required here as this is internal plugin implementation and plugin works on already validated nodes. For operators this function is public API.
It also look this check are not done in template plugin.

Reply:
Okay, then probably size check is not necessary, but adding  `set_shape` is still valid.

Reply:
Are you sure? This is used for data inference so all shapes should be set.

Reply:
I've added both checks, even if they are not necessary, they are really simple to implement. If any of those checks shouldn't be implemented please let me know, I'll delete it